docs/pages/CodeAnalysis.md

```diff
@@ -0,0 +1,226 @@
+
+### Component Analyzer
+
+The [Component Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/ComponentAnalyzer.js) analyzes JavaScript files named `Component.js` to collect dependency information by searching for a `manifest.json` located in the same folder as the `Component.js`. If one is found, the `sap.ui5` section is evaluated in the following way:
+
+- Any library dependency is added as a dependency to the `library.js` module of that library. If the library dependency is modeled as 'lazy', the
+ module dependency will be added as 'conditional'.
+- Any component dependency is added as a dependency to a file named `Component.js`. If the dependency is modeled as 'lazy', the module dependency will be added as 'conditional'.
+- For each UI5 model for which a type is configured, a module dependency to that type is added.
+- For each route that contains a view name, a module dependency to that view is added.
+
+### Fiori Elements Analyzer
+
+The [Fiori Elements Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/FioriElementsAnalyzer.js) analyzes a FioriElements app and its underlying template components to collect dependency information. It searches for a `manifest.json` located in the same folder as the `Component.js`. If one is found the `sap.fe` section will be evaluated in the following way:
+
+- For each entity set in the `entitySets` object, each sub-entry is checked for a `default.template` property.
+- If one is found, the containing string is interpreted as the short name of a template component in the `sap.fe.templates` library.
+- A dependency to that template component is added to the analyzed app.
+
+For a full analysis, "routing" also should be taken into account. Only when a sub-entry of the entity set
+is referenced by a route, the template for that entry will be used. Routes thereby could form entry points.
+
+```json
+{
+    "sap.fe" : {
+        "entitySets" : {
+            "C_AIVS_MDBU_ArtistTP" : {
+                "feed": {
+                    "default": {
+                        "template": "ListReport"
+                    }
+                },
+                "entry" : {
+                    "default" : {
+                        "outbound" : "musicV2Display"
+                    }
+                }
+            }
+        },
+        "routing" : {
+            "routes" :{
+                "ArtistList": {
+                    "target": "C_AIVS_MDBU_ArtistTP/feed"
+                }
+            }
+        }
+    }
+}
+```
+
+The template component is analyzed in the following way:
+
+- precondition: the template component class is defined in an AMD-style module, using define or sap.ui.define
+- precondition: the module 'sap/fe/core/TemplateAssembler' is imported
+- precondition: a call to TemplateAssembler.getTemplateComponent is used to define the component class
```

Comment:
--> wrap TemplateAssembler.getTemplateComponent in backticks

Reply:
done